microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
 
-_Renderer memory_ is all memory of the process where an inspected page is rendered:
+The JavaScript heap is part of the renderer memory. The renderer memory is the memory used by the browser process where the webpage is rendered. The renderer memory is comprised of the following:
 
-_renderer memory_ = _native memory_ + _JS heap memory of the page_ + _JS heap memory of all dedicated workers started by the page_
+* Native memory (such as the one used by the C++ objects that represent the DOM nodes).
+* JavaScript heap memory of the page.
+* JavaScript heap memory of all dedicated workers started by the page.
 
-Nevertheless, even a small object can hold a large amount of memory indirectly, by preventing other objects from being disposed of by the automatic garbage collection process.
 
-### Retained size
-
-The _retained size_ is the size of memory that is freed once the object is deleted along with the dependent objects that were made unreachable from garbage-collection roots (GC roots).
+<!-- ====================================================================== -->
+## Garbage-collection roots
 
-_Garbage-collection roots_ are made up of _handles_ that are created (as either local or global) when making a reference from native code to a JavaScript object outside of the V8 VM.  All such handles can be found within a heap snapshot under **GC roots** > **Handle scope** and **GC roots** > **Global handles**.  Describing the handles in this documentation without diving into details of the browser implementation may be confusing.  Both garbage-collection roots and the handles aren't something you need to worry about.
+_Garbage-collection roots_, or GC roots, are created by the browser when a reference from the browser's native code to a JavaScript object outside of the V8 virtual machine is made. These references are called _handles_. You can find these handles in the **Memory** tool under **GC roots** > **Handle scope** and **GC roots** > **Global handles**.<!-- Can't find this in the Memory tool. Where is it? -->
 
-There are many internal GC roots, most of which aren't interesting for the users.  From the applications standpoint, there are the following kinds of roots:
+There are many internal GC roots, most of which aren't interesting for web developers. From the webpage's standpoint, the following types of GC roots exist:
 
-*  Window global object (in each iframe).  In the heap snapshots, the `distance` field indicates the number of property references on the shortest retaining path from the window.
+*  Window global object (in each iframe). In a heap snapshot displayed in the **Memory** tool, the `distance` field for a window global object indicates the number of property references that on the shortest retaining path from the window.<!-- If this is useful, this needs clarification: what window? The top-level one? Does retaining path mean the path through the graph from the iframe window to the top-level window? -->
```

Comment:
@sethbrenith this sentence is hard to parse. I'm wondering if it's still useful, and if it is, would you have a suggestion for making it clearer?

Reply:
Agreed: this sentence seems low-value and sort of irrelevant to the primary topic of this section, which is describing the roots.

Reply:
Removed the sentence about the distance field.